src/Transport/Sending/Batcher.cs

```diff
@@ -10,6 +11,7 @@ public Batcher(ITopologySectionManagerInternal topologySectionManager, ReadOnlyS
         {
             this.topologySectionManager = topologySectionManager;
             messageSizePaddingPercentage = settings.Get<int>(WellKnownConfigurationKeys.Connectivity.MessageSenders.MessageSizePaddingPercentage);
+            localAddress = new Lazy<string>(() => settings.LocalAddress());
```

Comment:
I'm ok with that although it smells like our whole initialization chain in the transport infrastructure should be overhauled. Not important enough though to hold up the PR

Reply:
The constructor is invoked way too early and LocalAddress is not available. If there's a better way that would not require massive refactoring, I'm up for a separate PR where we could experiment.